testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/http/AbstractHttpServerTest.java

```diff
@@ -543,6 +543,49 @@ void requestWithNonStandardHttpMethod() throws InterruptedException {
     }
   }
 
+  @Test
+  void extractSingleBaggage() {
+    String method = "GET";
+    AggregatedHttpRequest request =
+        AggregatedHttpRequest.of(
+            request(SUCCESS, method).headers().toBuilder()
+                .set("baggage", "test-baggage-key-1=test-baggage-value-1")
```

Comment:
I think you should add a comment here that says `adding baggage header in w3c baggage format` or something like that.